OpenRA.Mods.Common/Projectiles/GravityBomb.cs

```diff
@@ -59,18 +59,23 @@ public GravityBomb(GravityBombInfo info, ProjectileArgs args)
 			this.info = info;
 			this.args = args;
 			pos = args.Source;
+			var actor = args.SourceActor;
 			var convertedVelocity = new WVec(info.Velocity.Y, -info.Velocity.X, info.Velocity.Z);
 			velocity = convertedVelocity.Rotate(WRot.FromFacing(args.Facing));
 			acceleration = new WVec(info.Acceleration.Y, -info.Acceleration.X, info.Acceleration.Z);
 
 			if (!string.IsNullOrEmpty(info.Image))
 			{
-				anim = new Animation(args.SourceActor.World, info.Image);
+				var facing = 0;
+				var ifacing = actor.TraitOrDefault<IFacing>();
+				if (ifacing != null)
+					facing = ifacing.Facing;
+				anim = new Animation(actor.World, info.Image, () => facing);
```

Comment:
What about doing it like this:
```
			var ifacing = actor.TraitOrDefault<IFacing>();  
			anim = new Animation(actor.World, info.Image, () => ifacing != null ? ifacing.Facing : 0);
```

Reply:
Won't this make it check every frame, while current one just checks at first and gives in the value.

Reply:
well it could be cached

Reply:
```
				var ifacing = actor.TraitOrDefault<IFacing>();  
				var facing = ifacing != null ? ifacing.Facing : 0;
				anim = new Animation(actor.World, info.Image, () => facing);
```

Reply:
well now it is just two lines less